### ISSUE TITLE
Giving a key for identification when using multiple ParentChildCheckbox

### ISSUE THREAD

**Issue by parjanyaacoder** [AbhishekDoshi26/parent-child-checkbox#1]
Hi, Abhishek Sir, 

I am very inspired by you for all your contributions in Flutter. I am also a Flutter Enthusiast. 

This issue is that each ParentChildCheckBox should have a key for its identification. Like instead of passing the selected children as a list we can pass the key to get a reference to the particular ParentChildCheckBox. 

ParentChildCheckBox(
key:'1',
) ,
ParentChildCheckBox(
key:'2',
).

Something like ParentChildCheckBox.key('1') to refer to the first ParentChildCheckbox(). 

I hope you understood. Correct me if I am wrong anywhere. 
Thank you.

**Comment by parjanyaacoder** [comment]
@AbhishekDoshi26

**Comment by AbhishekDoshi26** [comment]
Hi @parjanyaacoder if you see the structure of `selectedChildrens` it works in the same way. Parent text becomes the key of the Checkbox using which you will get the list of all selected children.

**Comment by parjanyaacoder** [comment]
Yes I agree. selectedChildrens is a map in which the parentText is the key for all selected childrens. I only presented a use case there might be other use cases for the same. A key for a ParentChildCheckBox might be helpful in upcoming versions of the package when you add more features. Also what about Nested ParentChildCheckbox ? Using a ParentChildCheckBox as one of the children of some other ParentChildCheckBox ?

**Comment by AbhishekDoshi26** [comment]
That's a really great point for nested one
Keeping this issue open for future reference @parjanyaacoder

**Comment by parjanyaacoder** [comment]
Thank you for appreciation. I am not very skilled in Flutter but will definitely try for a pr.

**Comment by AbhishekDoshi26** [comment]
Adding `PR Welcomed` label for reference.

**Comment by DavidWoli** [comment]
@parjanyaacoder @AbhishekDoshi26 Is someone working on this issue? If yes, I won't touch it.

**Comment by AbhishekDoshi26** [comment]
@DavidWoli no one is working. Feel free to assign it to yourself :)

Thanks!

**Comment by DavidWoli** [comment]
@AbhishekDoshi26 Okay. This week I'm studying hard for some programming exams, so I might be a bit busy. However, I should have more free time over the weekend and could potentially work on a PR then. 

Thanks for your understanding!

**Comment by AbhishekDoshi26** [comment]
> @AbhishekDoshi26 Okay. This week I'm studying hard for some programming exams, so I might be a bit busy. However, I should have more free time over the weekend and could potentially work on a PR then.
> 
> Thanks for your understanding!

Sure thing. Not an issue. Whenever you get time :)

**Comment by AbhishekDoshi26** [comment]
@DavidWoli are you working on this?
